### PR TITLE
ddl: make test TestRenameTableAutoIDs more stable

### LINKED ISSUE #53765: flaky test TestRenameTableAutoIDs
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 3. What did you see instead (Required)

```
     testkit.go:157: 
        	Error Trace:	pkg/testkit/testkit.go:157
        	            				pkg/testkit/testkit.go:150
        	            				pkg/ddl/schema_test.go:398
        	Error:      	Received unexpected error:
        	            	[schema:1146]Table 'renametableautoids2.t2' doesn't exist
        	            	github.com/pingcap/errors.AddStack
        	            		external/com_github_pingcap_errors/errors.go:178
        	            	github.com/pingcap/errors.Trace
        	            		external/com_github_pingcap_errors/juju_adaptor.go:15
        	            	github.com/pingcap/tidb/pkg/planner/core.Preprocess
        	            		pkg/planner/core/preprocess.go:141
        	            	github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile
        	            		pkg/executor/compiler.go:67
        	            	github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
        	            		pkg/session/session.go:2094
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).ExecWithContext
        	            		pkg/testkit/testkit.go:383
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).MustExecWithContext
        	            		pkg/testkit/testkit.go:155
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).MustExec
        	            		pkg/testkit/testkit.go:150
        	            	github.com/pingcap/tidb/pkg/ddl_test.TestRenameTableAutoIDs
        	            		pkg/ddl/schema_test.go:398
        	            	testing.tRunner
        	            		GOROOT/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		src/runtime/asm_amd64.s:1650
        	Test:       	TestRenameTableAutoIDs
        	Messages:   	sql:insert into RenameTableAutoIDs2.t2 values (50, 5), [], error stack [schema:1146]Table 'renametableautoids2.t2' doesn't exist
        	            	github.com/pingcap/errors.AddStack
        	            		external/com_github_pingcap_errors/errors.go:178
        	            	github.com/pingcap/errors.Trace
        	            		external/com_github_pingcap_errors/juju_adaptor.go:15
        	            	github.com/pingcap/tidb/pkg/planner/core.Preprocess
        	            		pkg/planner/core/preprocess.go:141
        	            	github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile
        	            		pkg/executor/compiler.go:67
        	            	github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
        	            		pkg/session/session.go:2094
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).ExecWithContext
        	            		pkg/testkit/testkit.go:383
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).MustExecWithContext
        	            		pkg/testkit/testkit.go:155
        	            	github.com/pingcap/tidb/pkg/testkit.(*TestKit).MustExec
        	            		pkg/testkit/testkit.go:150
        	            	github.com/pingcap/tidb/pkg/ddl_test.TestRenameTableAutoIDs
        	            		pkg/ddl/schema_test.go:398
        	            	testing.tRunner
        	            		GOROOT/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		src/runtime/asm_amd64.s:1650 
```

https://tiprow.hawkingrei.com/view/gs/pingcapprow/pr-logs/pull/pingcap_tidb/53762/fast_test_tiprow/1797589167828373504

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #53765

Problem Summary:

### What changed and how does it work?

The root cause is the same with this one https://github.com/pingcap/tidb/pull/54447
When we do ddl, wait ddl status change to finish is not enough.
We need to wait the infoschema reload finish, otherwise we may see the stale infoschema.

Here the test code execute `renamte table  renametableautoids.t   to  renametableautoids2.t2`
Using the stale infoschema would get error :

```
        	Error:      	Received unexpected error:
        	            	[schema:1146]Table 'renametableautoids2.t2' doesn't exist
        	            	github.com/pingcap/errors.AddStack
        	            		external/com_github_pingcap_errors/errors.go:178
```

NOTE, rename table cross different DBs have known capatibility issue with  autoid.
And even after this fix, the test may still be unstable by that capability issue.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/schema_test.go [hunk 1]
```diff
@@ -412,19 +413,38 @@ func TestRenameTableAutoIDs(t *testing.T) {
...
 	go func() {
 		alterChan <- tk1.ExecToErr(`rename table t to ` + dbName + `2.t2`)
 	}()
 	waitFor(11, "t", "running")
 	waitFor(4, "t", "public")
+
+	// ddl finish does not mean the infoschema loaded.
+	// when infoschema v1->v2 switch, it take more time, so we must wait to ensure
+	// the new infoschema is used.
+	for {
```
Comment (xhebox) [inline pkg/ddl/schema_test.go:428]: Maybe we can use `require.Eventually` and add a final timeout?

Comment (tiancaiamao) [reply pkg/ddl/schema_test.go]: Done

**Review by xhebox — APPROVED** [review]

**Review by mjonss — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**